usecases/scheduled_execution/offloading_job.go

```diff
@@ -0,0 +1,212 @@
+	return river.NewPeriodicJob(
+		river.PeriodicInterval(interval),
+		func() (river.JobArgs, *river.InsertOpts) {
+			return models.OffloadingArgs{
+					OrgId: orgId,
+				}, &river.InsertOpts{
+					Queue: orgId,
+					UniqueOpts: river.UniqueOpts{
+						ByQueue:  true,
+						ByPeriod: interval,
+						ByState: []rivertype.JobState{
+							rivertype.JobStateAvailable,
+							rivertype.JobStatePending, rivertype.JobStateRunning,
+							rivertype.JobStateRetryable, rivertype.JobStateScheduled,
+						},
+					},
+				}
+		},
+		&river.PeriodicJobOpts{RunOnStart: true},
+	)
+}
+
+type OffloadingWorker struct {
+	river.WorkerDefaults[models.OffloadingArgs]
+
+	executorFactory    executor_factory.ExecutorFactory
+	transactionFactory executor_factory.TransactionFactory
+	repository         offloadingRepository
+	blobRepository     repositories.BlobRepository
+
+	config infra.OffloadingConfig
+}
+
+func NewOffloadingWorker(executorFactory executor_factory.ExecutorFactory, transactionFactory executor_factory.TransactionFactory,
+	repository offloadingRepository, blobRepository repositories.BlobRepository,
+	offloadingBucketUrl string, offloadConfig infra.OffloadingConfig,
+) *OffloadingWorker {
+	return &OffloadingWorker{
+		executorFactory:    executorFactory,
+		transactionFactory: transactionFactory,
+		repository:         repository,
+		blobRepository:     blobRepository,
+		config:             offloadConfig,
+	}
+}
+
+func (w OffloadingWorker) Work(ctx context.Context, job *river.Job[models.OffloadingArgs]) error {
+	logger := utils.LoggerFromContext(ctx)
+	exec := w.executorFactory.NewExecutor()
+	timeout := time.After(w.config.JobInterval - 5*time.Minute)
+	limiter := time.NewTicker(time.Second / time.Duration(w.config.WritesPerSecond))
+	defer limiter.Stop()
+
+loop:
+	for {
+		wm, err := w.repository.GetOffloadingWatermark(ctx, exec, job.Args.OrgId, "decision_rules")
+		if err != nil {
+			return err
+		}
+
+		req := models.OffloadDecisionRuleRequest{
+			OrgId:        job.Args.OrgId,
+			DeleteBefore: time.Now().Add(-w.config.OffloadBefore),
+			BatchSize:    w.config.BatchSize,
+			Watermark:    wm,
+		}
+
+		rules, err := w.repository.GetOffloadableDecisionRules(ctx, w.executorFactory.NewExecutor(), req)
+		if err != nil {
+			return err
+		}
+
+		offloadedIds := make([]string, w.config.SavepointEvery)
+		idx := 0
+
+		var lastOfBatch *models.OffloadableDecisionRule
+
+		for item := range rules {
+			select {
+			case <-limiter.C:
+			case <-timeout:
+				break loop
+			}
+
+			if item.Error != nil {
+				return item.Error
+			}
+
+			rule := item.Model
+
+			if rule.RuleExecutionId == nil {
+				lastOfBatch = &rule
+				continue
+			}
+
+			if rule.RuleEvaluation != nil {
+				key := w.repository.GetOffloadedDecisionRuleKey(req.OrgId,
+					rule.DecisionId, *rule.RuleId, rule.CreatedAt)
+
+				opts := blob.WriterOptions{Metadata: map[string]string{
+					"Custom-Date": rule.CreatedAt.Format(time.RFC3339),
```

Comment:
GCS-specific metadata attribute from which we can create custom lifecycle rules. There is no equivalent, that I know of, for AWS or Azure.